src/parsers/power_system_table_data.jl

```diff
@@ -827,21 +827,22 @@ function make_hydro_generator(data::PowerSystemTableData, gen, bus)
     )
 
     curtailcost = 0.0
-    storage_df = data.category_to_df[STORAGE]
-    storage_capacity = nothing
-    initial_storage = nothing
-    inflow = nothing
-    for row in eachrow(storage_df)
-        row[Symbol("GEN UID")], gen.name
-        if row[Symbol("GEN UID")] == gen.name
-            storage_capacity = row[Symbol("Max Volume GWh")]
-            initial_storage = row[Symbol("Initial Volume GWh")]
-            inflow = row[Symbol("Inflow Limit GWh")]
-            break
+    if gen_type == HydroDispatch
+        @debug("Creating $(gen.name) as HydroDispatch")
+        !haskey(data.category_to_df, STORAGE) && throw(DataFormatError("Storage information must defined in storages csv"))
+        storage_df = data.category_to_df[STORAGE]
+        storage_capacity = nothing
+        initial_storage = nothing
+        inflow = nothing
+        for row in eachrow(storage_df)
+            row[Symbol("GEN UID")], gen.name
+            if row[Symbol("GEN UID")] == gen.name
+                storage_capacity = row[Symbol("Max Volume GWh")]
+                initial_storage = row[Symbol("Initial Volume GWh")]
+                inflow = row[Symbol("Inflow Limit GWh")]
+                break
+            end
```

Comment:
After the for loop you should throw an error if isnothing(storage_capacity).

Reply:
I do agree with that